packages/esm-bed-management-app/src/bed-administration/form/add-edit-bed.resource.ts

```diff
@@ -0,0 +1,102 @@
+import { useMemo } from 'react';
+import useSWR from 'swr';
+import { type FetchResponse, openmrsFetch, restBaseUrl } from '@openmrs/esm-framework';
+import { type BedPostPayload, type BedTag } from '../../types';
+
+interface BedForm {
+  bedNumber: string;
+  bedType: string;
+  row: number;
+  column: number;
+  status: string;
+  locationUuid: string;
+  uuid?: string;
+}
+
+export async function saveBed({ bedPayload }: { bedPayload: BedPostPayload }): Promise<FetchResponse<BedForm>> {
+  const response = await openmrsFetch(`${restBaseUrl}/bed`, {
+    method: 'POST',
+    headers: { 'Content-Type': 'application/json' },
+    body: {
+      bedNumber: bedPayload.bedNumber,
+      bedType: bedPayload.bedType,
+      row: bedPayload.row,
+      column: bedPayload.column,
+      status: bedPayload.status,
+      locationUuid: bedPayload.locationUuid,
+    },
+  });
+
+  if (response.status === 201 && bedPayload.bedTag && bedPayload.bedTag.length > 0) {
+    const bedUuid = response.data.uuid;
+    await createBedTagMappings(bedUuid, bedPayload.bedTag);
+  }
+
+  return response;
+}
+
+export async function editBed({
+  bedPayload,
+  bedId,
+}: {
+  bedPayload: BedPostPayload;
+  bedId: string;
+}): Promise<FetchResponse<BedForm>> {
+  const response = await openmrsFetch(`${restBaseUrl}/bed/${bedId}`, {
+    method: 'POST',
+    headers: { 'Content-Type': 'application/json' },
+    body: {
+      uuid: bedPayload.uuid,
+      bedNumber: bedPayload.bedNumber,
+      bedType: bedPayload.bedType,
+      row: bedPayload.row,
+      column: bedPayload.column,
+      status: bedPayload.status,
+      locationUuid: bedPayload.locationUuid,
+    },
+  });
+
+  if (response.status === 200 || response.status === 201) {
+    await updateBedTagMappings(bedId, bedPayload.bedTag || []);
+  }
```

Comment:
We need a way to clean up existing tags so we don't end up with duplicate and unwanted tags. So when effecting an update, we'd need to delete all existing bed tag mappings for a bed and then create new mappings for the new tags. Maybe using something like this:


```suggestion
  try {
    const existingMappingsResponse = await openmrsFetch(`${restBaseUrl}/bedTagMap?v=full`);
    const existingMappings =
      existingMappingsResponse.data?.results?.filter((mapping: BedTagMapResponse) => mapping.bed?.uuid === bedUuid) ||
      [];

    const deletePromises = existingMappings.map((mapping: BedTagMapResponse) =>
      openmrsFetch(`${restBaseUrl}/bedTagMap/${mapping.uuid}`, {
        method: 'DELETE',
      }),
    );
    await Promise.all(deletePromises);

    await createBedTagMappings(bedUuid, newBedTags);
  } catch (error) {
    console.error('Error updating bed tag mappings:', error);
    await createBedTagMappings(bedUuid, newBedTags);
  }
```